parlai/agents/transformer/modules.py

```diff
@@ -310,17 +349,23 @@ def forward(self, input):
         if self.embeddings_scale:
             tensor = tensor * np.sqrt(self.dim)
         tensor = tensor + self.position_embeddings(positions).expand_as(tensor)
+        if self.embedding_normalization:
+            tensor = self.layer_norm_emb(tensor)
         # --dropout on the embeddings
         tensor = self.dropout(tensor)
 
         tensor *= mask.unsqueeze(-1).float()
         for i in range(self.n_layers):
             tensor = self.layers[i](tensor, mask)
-
         if self.reduction:
-            divisor = mask.float().sum(dim=1).unsqueeze(-1).clamp(min=1e-20)
-            output = tensor.sum(dim=1) / divisor
-            return output
+            if self.reduction_type == 'first':
+                return tensor[:, 0, :]
+            elif self.reduction_type == 'max':
+                return tensor.max(dim=1)[0]
+            else:
```

Comment:
I would prefer an explicit elif for `mean` here, and then the else would raise a ValueError